clients/rust/tests/deactivate_stake.rs

```diff
@@ -0,0 +1,493 @@
+    let updated_config = Account {
+        lamports: account.lamports,
+        data: config_account.try_to_vec().unwrap(),
+        owner: account.owner,
+        executable: account.executable,
+        rent_epoch: account.rent_epoch,
+    };
+    context.set_account(&config, &updated_config.into());
+
+    let validator = Pubkey::new_unique();
+    let authority = Keypair::new();
+    let vote = create_vote_account(&mut context, &validator, &authority.pubkey()).await;
+
+    // And a stake account.
+
+    let stake_pda = create_stake(&mut context, &validator, &vote, &config).await;
+
+    let account = get_account!(context, stake_pda);
+    let mut stake_account = Stake::from_bytes(account.data.as_ref()).unwrap();
+    // "manually" set the amount to 100
+    stake_account.amount = 100;
+
+    let updated_stake = Account {
+        lamports: account.lamports,
+        data: stake_account.try_to_vec().unwrap(),
+        owner: account.owner,
+        executable: account.executable,
+        rent_epoch: account.rent_epoch,
+    };
+
+    context.set_account(&stake_pda, &updated_stake.into());
+
+    // And we deactivate an amount from the stake account.
+
+    let deactivate_ix = DeactivateStakeBuilder::new()
+        .config(config)
+        .stake(stake_pda)
+        .stake_authority(authority.pubkey())
+        .amount(5)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[deactivate_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &authority],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    let account = get_account!(context, stake_pda);
+    let stake_account = Stake::from_bytes(account.data.as_ref()).unwrap();
+    assert_eq!(stake_account.deactivating_amount, 5);
+
+    let mut clock = context.banks_client.get_sysvar::<Clock>().await.unwrap();
+    // timestamp of the first deactivation
+    let timestamp = clock.unix_timestamp;
+    clock.unix_timestamp = timestamp.saturating_add(1000);
+    context.set_sysvar::<Clock>(&clock);
+
+    // When we deactivate a different amount from the stake account
+    // with an active deactivation.
+
+    let deactivate_ix = DeactivateStakeBuilder::new()
+        .config(config)
+        .stake(stake_pda)
+        .stake_authority(authority.pubkey())
+        .amount(1)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[deactivate_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &authority],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    // Then the deactivation should be "updated".
+
+    let account = get_account!(context, stake_pda);
+    let stake_account = Stake::from_bytes(account.data.as_ref()).unwrap();
+    assert_eq!(stake_account.deactivating_amount, 1);
+    assert!((timestamp as u64) < stake_account.deactivation_timestamp.value().unwrap());
```

Comment:
Why not just create a variable with the advanced timestamp value (ie. `timestamp + 1000`), and do a direct comparison here?

Reply:
Sorry, I don't follow your comment. What would this variable replace?

Reply:
You're checking that the timestamp you first set on the stake account is less than the timestamp that's currently on the stake account. The check is okay, I'm just suggesting being a bit more explicit about checking the stake account's value.

```rust
let original_timestamp = clock.timestamp;
let new_timestamp = original_timestamp.saturating_add(1000);
clock.timestamp = new_timestamp;
/* ... */
assert_eq!(stake_account.timestamp, new_timestamp);
```

Reply:
Got it, makes sense. I updated the test.